### ISSUE TITLE
Exception thrown using HttpClient with handler

### ISSUE THREAD

**Issue by jskeet** [GoogleCloudPlatform/google-cloud-dotnet#792]
I have an HttpClient which is meant to propagate trace information:

```csharp
var handler = TraceHeaderPropagatingHandler.Create(tracer);
httpClient = new HttpClient(handler);
```

Unfortunately when I call `GetStringAsync` I get an exception:

```text
System.InvalidOperationException occurred
  HResult=-2146233079
  Message=The inner handler has not been assigned.
  Source=System.Net.Http
  StackTrace:
       at System.Net.Http.DelegatingHandler.SetOperationStarted()
       at System.Net.Http.DelegatingHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
       at Google.Cloud.Diagnostics.Common.TraceHeaderPropagatingHandler.<SendAsync>d__3.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at System.Net.Http.HttpClient.<FinishSendAsync>d__58.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at System.Net.Http.HttpClient.<GetContentAsync>d__32`1.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
       at Google.NextTalk.Web.Backends.JsonBackend.<GetLanguagesAsync>d__6.MoveNext() in
```

**Comment by jskeet** [comment]
(It's just possible that this is due to me using `Result` from the async result, which isn't ideal... will fix that and see if it helps, although I suspect not.)

**Comment by jskeet** [comment]
Hmm... the same code in an integration test is fine. I'm very confused at this point.

**Comment by jskeet** [comment]
Aha - it was fixed in 7e0191ebb7a4f5e4ede3bc99adf4d24d0dfae58f.
New release should sort it :)
